symfit/contrib/interactive_guess/interactive_guess.py

```diff
@@ -8,62 +8,86 @@
-        a red line.
-        
+        Data points are plotted as a blue contour plot, the proposed model as
+        a red line. The errorbars on the proposed model represent the
+        percentile of data within the thresholds.
+
         Slider extremes are taken from the parameters where possible. If
         these are not provided, the minimum is 0; and the maximum is value*2.
         If no initial value is provided, it defaults to 1.
 
         This will modify the values of the parameters present in model.
 
         :param n_points: The number of points used for drawing the
-            fitted function.
+            fitted function. Defaults to 50.
         :type n_points: int
-        """
-        n_points = kwargs.pop('n_points')
-        super(InteractiveGuess2D, self).__init__(*args, **kwargs)
 
-        if len(self.independent_data) != 1:
-            raise IndexError("Only 2D problems are supported.")
+        :param log_contour: Whether to plot the contour plot of the log of the
+            density, rather than the density itself. If True, any density less
+            than 1e-7 will be considered 0. Defaults to True.
+        :type log_contour: bool
 
+        :param percentile: Controls the errorbars on the proposed model, such
+            that the lower errorbar will cover percentile[0]% of the data, and
+            the upper will cover percentile[1]%. Defaults to [5, 95], with
+            corresponds to a 90% percentile. Should be a list of 2 numbers.
+        :type percentile: list
+        """
+        self.log_contour = kwargs.pop('log_contour')
+        n_points = kwargs.pop('n_points')
+        self.percentile = kwargs.pop('percentile')
+        super(InteractiveGuess, self).__init__(*args, **kwargs)
+        if len(self.independent_data) > 1:
+            self._dimension_strategy = StrategynD(self)
+        else:
+            self._dimension_strategy = Strategy2D(self)
+        # TODO: Some of the code here is specific to the n-D case and should
+        # be moved.
         self._projections = list(itertools.product(self.model.independent_vars,
                                                    self.model.dependent_vars))
         x_mins = {v: np.min(data) for v, data in self.independent_data.items()}
         x_maxs = {v: np.max(data) for v, data in self.independent_data.items()}
 
-        # Stretch the plot 10-20% in the X direction, since that is visually more
-        # appealing. We can't evaluate the model for x < x_initial, so don't.
+        # Stretch the plot 10-20% in the X direction, since that is visually
+        # more appealing. We can't evaluate the model for x < x_initial, so
+        # don't.
         for x in self.model.independent_vars:
             plotrange_x = x_maxs[x.name] - x_mins[x.name]
-            if not hasattr(self.model, 'initial'):
+            # if not hasattr(self.model, 'initial'):
+            if not isinstance(self.model, ODEModel):
```

Comment:
why treat ODEModel seperate here? I'm not sure they have to be here, but a comment is missing.